METHOD_NAMING_PROPOSAL.md

```diff
@@ -0,0 +1,412 @@
+# v5 Method Naming Proposal
+
+## Overview
+
+In v5 of the Braintree Android SDK, we are introducing a new integration pattern that includes a 
+separate `Launcher` class for each payment method that requires launching a browser, external 
+app, payment sheet, etc. to complete the payment flow. The addition of this pattern requires new 
+method and result object names, and v5 provides an opportunity to revisit the existing naming 
+patterns across payment methods.
+
+## Integration Comparison
+
+This document provides alternative naming options in comparison to the v4 integration to 
+determine the best pattern for v5. 
+
+In v4, most payment flows require one method call (ex: `tokenize`), and the browser/app switch is handled under the hood by the SDK, before returning a `PaymentMethodNonce` to the merchant's listener. Some payment methods require two method calls (ex: 3DS `performVerification` -> `continuePerformVerification`).
+
+In v5, all app/broswer switching payment methods require starting the payment flow by invoking a method on the `Client` to retrieve data to be passed to the `Launcher`. With that data, the `Launcher` presents the payment sheet, launches the browser, or launches the exteral app for some user interaction/authentication/payment authorization. The `Launcher` will be instantiated with a callback that will be invoked upon return to the merchant app from the browser/external app/payment sheet. Within the callback, another method on the `Client` will be invoked to handle data from the app/browser switch result and return a `PaymentMethodNonce`. 
+
+### Minimal Change
+
+With this implementation, the method names from v4 are maintained where possible. The comments above each method show what the method actually does, as the naming doesn't entirely align with this approach.
+
+<table>
+<tr>
+<th>v4 PayPal</th>
+<th>v5 PayPal</th>
+</tr>
+<tr>
+
+<td>
+
+```kotlin 
+// Retrieve data required to launch browser
+// Launch browser flow, parse browser result
+// Return nonce to listener
+payPalClient.tokenizePayPalAccount(activity, request)
+```
+
+</td>
+
+<td>
+
+```kotlin
+payPalLauncher = PayPalLauncher() { payPalBrowserSwitchResult ->
```

Comment:
To confirm, this pattern would require a `launcher` object for each individual payment method, correct? A `launcher` object cannot and should not be shared across payment method flows?

Reply:
I was thinking about the launcher too, Holly. My understanding is that it will be separate for each payment method and abstracts away details of laughing webView or app switch. Except card module?

Reply:
Yes it needs to be a separate launcher for each payment method because that class is responsible for actually launching the flow (i.e. opening PayPal browser or switching to Venmo app), so it is payment type specific.